skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkateService.kt

```diff
@@ -36,14 +36,21 @@ interface SkateProjectService {
 class SkateProjectServiceImpl(private val project: Project) : SkateProjectService {
 
   override fun showWhatsNewWindow() {
-    // TODO
-    //  Only show when changed
-    //  Only show latest changes
+
     val settings = project.service<SkatePluginSettings>()
+    val changelogJournal = project.service<ChangelogJournal>()
+
     if (!settings.isWhatsNewEnabled) return
     val projectDir = project.guessProjectDir() ?: return
+
     val changeLogFile = VfsUtil.findRelativeFile(projectDir, settings.whatsNewFilePath) ?: return
     val changeLogString = VfsUtil.loadText(changeLogFile)
+
+    // Don't show the tool window if the parsed changelog is blank
+    val parsedChangelog = ChangelogParser.readFile(changeLogString, changelogJournal.lastReadDate)
```

Comment:
Shouldn't we be only passing this in to the window below? Looks like we're still passing in the full changelogstring

Reply:
My logic here is to check if the `parsedChangelog` would be empty, because if it is, it will return before the creation of the tool window. So I put it before

Reply:
Right but isn't the window always going to show the whole `changeLogString` each time currently, even if we only want to show a partial of that?

Reply:
I think right now it just shows the `parsedChangelog` and I'm passing in the `changeLogString `to parse it, right? Do you mean that the change I'm making is always going to display the full string?